docs/source/models.rst

```diff
@@ -307,49 +236,136 @@ model artifacts, together with other model metadata. To set a signature on a log
-optional string column.
-The output is an unnamed integer specifying the predicted class.
+Column-based Signature
+~~~~~~~~~~~~~~~~~~~~~~
+This signature is typically used for traditional machine learning models that take tabular data as input such as Pandas 
+DataFrame. Column-based schemas are a sequence of (optionally) named columns with data type. Each column-based input and 
+output is represented by a type corresponding to one of :ref:`supported data types <supported-data-types-column>` and an 
+optional name. Input columns can also be marked as ``optional``, indicating whether they are required as input to the 
+model or can be omitted. 
+
+The following example displays the model signature for a classification model trained on the 
+`Iris dataset <https://archive.ics.uci.edu/ml/datasets/iris>`_. The input has 4 named, numeric columns and 
+1 named, optional string column. The output is an unnamed integer specifying the predicted class.
 
 .. code-block:: yaml
 
   signature:
-      inputs: '[{"name": "sepal length (cm)", "type": "double"}, {"name": "sepal width
-        (cm)", "type": "double"}, {"name": "petal length (cm)", "type": "double"}, {"name":
-        "petal width (cm)", "type": "double"}, {"name": "class", "type": "string", "optional": "true"}]'
+      inputs: '[
+        {"name": "sepal length (cm)", "type": "double"}, 
+        {"name": "sepal width(cm)",   "type": "double"}, 
+        {"name": "petal length (cm)", "type": "double"}, 
+        {"name": "petal width (cm)",  "type": "double"}, 
+        {"name": "class",             "type": "string",  "optional": "true"}
+      ]'
       outputs: '[{"type": "integer"}]'
       params: null
 
-Tensor-based Signature Example
-""""""""""""""""""""""""""""""
+.. _supported-data-types-column:
+
+Supported Data Types
+""""""""""""""""""""
+Column-based signature supports :py:class:`MLflow DataType <mlflow.types.DataType>` that defines following data primitives,
+
+* string
+* integer
+* boolean
+* double
+* float
+* long
+* datetime
+
+.. code-block:: python
+
+    from mlflow.models import infer_signature
+
+    sign = infer_signature(model_input={"long_col": 1, "str_col": "a", "bool_col": True})
+    print(sign.input)
+    # [
+    #   'long_col': long (required),
+    #   'str_col': string (required),
+    #   'bool_col': boolean (required)
+    # ]
+
+    ## You can also infer signature from Pandas DataFrame. Column with None value will be `optional`
+    sign = infer_signature(model_input=pd.DataFrame({"col": [1.0, 2.0, None]}))
+    print(sign.input)
+    # [ 'col': long (optional) ]
+
+
+Column-based signature also support composite data types of these primitives,
+
+* Array (list, numpy arrays)
+* Object (dictionary)
+
+.. code-block:: python
+
+    # Array (from list)
+    sign = infer_signature(model_input={"list_col": ["a", "b", "c"]})
+    print(sign.input)
+    # [ 'list_col': Array(string) (required) ]
+
```

Comment:
I'm thinking that what we want to highlight here might be the result `[ 'list_col': Array(string) (required) ]` instead of the code 🤔 I feel like above yaml part of showing signature example is much clearer than this. WDYT? Though this is not a blocker, It explains what we want to show, just wondering what people thinks about it :D

Reply:
Yeah I agree that yaml is the best way to print schema and comment is a bit ugly...
One challenge is that we still want to show python code for input sample. Ideally we can show python/yaml side-by-side with nice style, but currently it doesn't work nicely.

Reply:
What about a table showing 'inputs' and 'inferred schema' ?

Reply:
Table works nicely indeed.

![Screenshot 2023-11-15 at 16 57 44](https://github.com/mlflow/mlflow/assets/31463517/25fdbc4e-cd05-4fbb-adb7-0ce3420a22ec)

A few pain points exist, but probably the best we can do for now (hopefully it becomes easier after md migration lol)
1. Only for single row apprently, if there is more than one, it gives gray background that doesn't look great. 
2. Super annoying to write .rst table with code block. It's fine for a few samples, but definitely not > 10.

<img width="1127" alt="Screenshot 2023-11-15 at 17 17 02" src="https://github.com/mlflow/mlflow/assets/31463517/897134e7-89fb-4cd2-bc4c-75709f22a3f3">

Reply:
Thanks for exploring this!! Though I was simply referring to text in the table lol, but this definitely looks better. We could also keep the current one migrate to tables afterwards :)